### PR TITLE
feat: Add support for reading ClientLibrarySettings from service configuration YAML

### PR DESCRIPTION
**Pull request by parthea**
Fixes b/356854861

### CODE REVIEW

File: gapic/schema/api.py [hunk 1]
```diff
@@ -670,6 +678,63 @@ def enforce_valid_method_settings(
...
+                library settings to be used when generating API methods.
+        Return:
+            None
+        Raises:
+            ClientLibrarySettingsError: if fields in `client_library_settings.experimental_features`
+                are not supported.
+        """
+
+        all_errors: dict = {}
+        versions_seen = []
```
Comment (ohmayr) [inline gapic/schema/api.py:727]: Can we make this a set?

```
versions_seen = {}
```

File: gapic/schema/api.py [hunk 2]
```diff
@@ -670,6 +678,63 @@ def enforce_valid_method_settings(
...
 
+    @cached_property
+    def all_library_settings(
+        self,
+    ) -> Mapping[str, Sequence[client_pb2.ClientLibrarySettings]]:
+        """Return a map of all `google.api.client.ClientLibrarySettings` to be used
+        when generating client libraries.
+        https://github.com/googleapis/googleapis/blob/master/google/api/client.proto#L130
+
+        Return:
```
Comment (ohmayr) [inline gapic/schema/api.py:689]: nit: `Returns`.
Same for the other function.

**Review by ohmayr — APPROVED** [review]

File: gapic/schema/api.py (hunk 2 above)
Comment (parthea) [reply gapic/schema/api.py]: Fixed in https://github.com/googleapis/gapic-generator-python/pull/2098/commits/e0d252e8cc474efd199bd6d8d1d44ec2806d997c

File: gapic/schema/api.py (hunk 1 above)
Comment (parthea) [reply gapic/schema/api.py]: Fixed in https://github.com/googleapis/gapic-generator-python/pull/2098/commits/d3c4b25845ce97a8d3d8422caac8596018b33dd0

**Pull request merged**